src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -1413,6 +2063,8 @@ private class TagTracker
             public SourceLocation TagLocation { get; }
 
             public SyntaxList<RazorSyntaxNode> PreviousNodes { get; }
+
+            public bool IsComplete { get; }
```

Comment:
This was necessary to keep track of because we assign different accepted characters based on whether a tag is well-formed or not.